Code/Framework/AzToolsFramework/AzToolsFramework/UI/Prefab/PrefabIntegrationManager.cpp

```diff
@@ -123,6 +117,13 @@ namespace AzToolsFramework
                 return;
             }
 
+            s_prefabFocusInterface = AZ::Interface<PrefabFocusInterface>::Get();
+            if (s_prefabFocusInterface == nullptr)
```

Comment:
suggestion: Instead of calling Get internally we might want to consider passing in these interfaces (maybe outside the scope of this PR) but will make things much easier to test in future. We're then not bound to the one global AZ::Interface instance (we can make setup easier by having a builder object if necessary that can return to us a constructed `PrefabIntegrationManager` for normal operation but can be customized while testing

Reply:
Would registering a different interface handler for testing purposes have the same effect?

PrefabIntegrationManager is a UI class that mostly gets tested via automation python tests, so I'm not sure such a refactor is necessary at the moment. I will be trying to add some unit tests to the focus mode as a separate task though, so if need be I will keep this recommendation in mind.

Reply:
Okay understood, sounds good 👍